platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ConfigValidator.java

```diff
@@ -39,37 +39,59 @@ static void validateMap(Map properties, String msg) {
         }
     }
 
-    static void validatePropertiesPath(Path propFile) {
-        if (propFile == null) {
-            throw new IllegalArgumentException("Null path provided");
-        }
-
-        String fileExtension = Files.getFileExtension(propFile.toString());
-        if (!(fileExtension.equalsIgnoreCase("props")
-                || fileExtension.equalsIgnoreCase("properties"))) {
-            throw new IllegalArgumentException(String.format("%s is not a properties file",
-                    propFile.toString()));
-        }
-
+    private static void protectHomeDir(Path target) {
         String ddfHomeProp = System.getProperty("ddf.home");
         validateString(ddfHomeProp, "No value set for system property ddf.home");
 
         Path ddfHomePath = Paths.get(ddfHomeProp);
-        if (!propFile.startsWith(ddfHomePath)) {
-            throw new IllegalArgumentException(String.format("File [%s] is not beneath ddf home directory [%s]",
-                    propFile.toString(),
+        if (!target.startsWith(ddfHomePath)) {
+            throw new IllegalArgumentException(String.format(
+                    "File [%s] is not beneath ddf home directory [%s]",
+                    target.toString(),
                     ddfHomePath.toString()));
         }
 
-        if (propFile.getParent()
+        if (target.getParent()
                 .equals(ddfHomePath)) {
             throw new IllegalArgumentException("Invalid attempt to edit file in ddf.home directory");
         }
+    }
 
-        if (propFile.getParent()
+    private static void protectEtcDir(Path target) {
+        if (target.getParent()
                 .endsWith("etc")) {
             throw new IllegalArgumentException(
                     "Invalid attempt to edit system file in /etc directory");
         }
     }
+
+    static void validateConfigPath(Path configFile) {
+        if (configFile == null) {
+            throw new IllegalArgumentException("Null path provided");
+        }
+
+        String fileExtension = Files.getFileExtension(configFile.toString());
+        if (!(fileExtension.equalsIgnoreCase("cfg") || fileExtension.equalsIgnoreCase("config"))) {
```

Comment:
Maybe consider configurable validation logic, where the caller passes along a case sensitivity flag. Because while validating that Java properties files are what they are should ignore case, it might make sense in this case not to ignore case. Worth a few moments thought, at least.